src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageDependencies.cs

```diff
@@ -150,6 +152,11 @@ internal ResolvePackageDependencies(LockFile lockFile, IPackageResolver packageR
         /// </summary>
         protected override void ExecuteCore()
         {
+            if (!EmitLegacyAssetsFileItems)
```

Comment:
"ResolvePackageDependencies doesn't need to be called at all unless EmitLegacyAssetsFileItems is true."